### PR TITLE
Feat/typescript types

### LINKED ISSUE #15: Add TypeScript typings
We use Flow internally, because that's what React Native comes with by default. Would be cool to add TS typings though :)

### PR DESCRIPTION
**Pull request by Esemesek**
Add Typescript types.
Fixes #15.

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { ReactTestInstance } from 'react-test-renderer';
+
+export interface GetByAPI {
+  getByName: (name: React.ReactType) => ReactTestInstance;
```
Comment (thymikee) [inline typings/index.d.ts]: `name: React.ReactType | string`, same for `getAllByName`

File: typings/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  getByText: (text: string | RegExp) => ReactTestInstance;
+  getByProps: (props: Record<string, any>) => ReactTestInstance;
+  getByTestId: (testID: string) => ReactTestInstance;
+  getAllByName: (name: React.ReactType) => Array<ReactTestInstance>;
+  getAllByText: (text: string | RegExp) => Array<ReactTestInstance>;
+  getAllByProps: (props: Record<string, any>) => Array<ReactTestInstance>;
+}
+
+export interface QueryByAPI {
+  queryByName: (name: React.ReactType) => ReactTestInstance | null;
```
Comment (thymikee) [inline typings/index.d.ts]: same as above

File: typings/index.d.ts (hunk 1 above)
Comment (Esemesek) [reply typings/index.d.ts]: `React.ReactType` already includes string C: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L52

Comment (thymikee) [reply typings/index.d.ts]: Oh, that's cool! Didn't know that :D

**Pull request merged**